### ISSUE TITLE
feat(components/icon): add title slot

### ISSUE THREAD

**Issue by chrisvaillancourt** [chrisvaillancourt/website#55]
## TLDR

Replace the name prop with a title slot.

## Summary

The [icon component](https://github.com/chrisvaillancourt/website/blob/af8e5b61955dd75368dd23e09c9979bd6b0f20d7/src/components/Icon.astro) uses a name prop to [populate the `<title>` element's content](https://github.com/chrisvaillancourt/website/blob/af8e5b61955dd75368dd23e09c9979bd6b0f20d7/src/components/Icon.astro#L38).

We should remove the name prop and add a title slot `<slot name="title">`. This approach makes it the component more declarative and allows us to only render the `<title>` if content is passed into the title slot.

**Comment by chrisvaillancourt** [comment]
Also look into refactoring how title works. 

For example, if we have an icon within a button, we want the title to appear when hovering the button instead of the icon within the button.
